refactor: Rename huddle to direct message group in email templates.

mreza-kiani opened the pull request:
This PR renames "huddle" references to "direct_message_group" in `locale/`.

Fixes the last piece of https://github.com/zulip/zulip/issues/28640

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

roanster007 commented on locale/ar/LC_MESSAGES/django.po:
```diff
@@ -1,7 +1,7 @@
 # Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
 # This file is distributed under the same license as the PACKAGE package.
-# 
+#
```
Leave these changes out of this commit.

mreza-kiani replied:
Updated.

roanster007 commented:
@mreza-kiani
i) you didn't update the corresponding `huddle_display_name` in  `templates/zerver/emails/missed_message.subject.txt`
ii) commit message might be better as `locale: Rename huddle to direct message group in locale files.`

mreza-kiani commented:
@roanster007 Thanks for the review.
i) Can you elaborate on what other references should be updated? Because I already updated [`missed_message.subject.txt`](https://github.com/zulip/zulip/pull/34107/files#diff-8f4b3e9f45cfd12dc6c853d9cdb48fae39a41c89b032900152a30fb4aca9ed03) and [`email_notification.py`](https://github.com/zulip/zulip/pull/34107/files#diff-a6ab658d945d49476471d429bc9c934a8d4ffa068bec866a3f6cd7bacbe71b48) that used this template variable.
ii) I updated the commit message to be more descriptive.

timabbott commented:
The locale changes might be useless with how Transifex syncing works, but they're probably harmless, and might save some utility. Merged, thanks @mreza-kiani!
